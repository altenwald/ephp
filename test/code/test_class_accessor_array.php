<?php

class A {
    public $array = [1,2,3,4];
    public $position = 0;

    public function get() {
        return $this->array[$this->position++];
    }
    public function get_error() {
        return $this->array["error"];
    }
}

$a = new A;
while ($data = $a->get()) print $data . "\n";
$a->get_error();

