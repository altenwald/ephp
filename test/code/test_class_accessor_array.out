1
2
3
4

Notice: Undefined offset 4 in {{CWD}}/test/code/test_class_accessor_array.php on line 8

Notice: Undefined index: error in {{CWD}}/test/code/test_class_accessor_array.php on line 11
